tests/test_trnsys.py

```diff
@@ -119,14 +127,597 @@ def test_get_save_write_schedules_as_input(self, config, converttest):
+        assert "!-LAYER " + materialNoMass[0].Name + "\n" in lines
+        assert "!-LAYER " + materials[0].Name + "\n" in lines
+
+    def test_relative_to_absolute(self, config, converttesteasy):
+        output_folder = None
+        (
+            idf,
+            idf_file,
+            weather_file,
+            window_lib,
+            trnsidf_exe,
+            template,
+            _,
+        ) = converttesteasy
+        try:
+            (
+                idf_file,
+                weather_file,
+                window_lib,
+                output_folder,
+                trnsidf_exe,
+                template,
+            ) = _assert_files(
+                idf_file, weather_file, window_lib, output_folder, trnsidf_exe, template
+            )
+        except:
+            output_folder = os.path.relpath(settings.data_folder)
+            print("Could not assert all paths exist - OK for this test")
+
+        # Check if cache exists
+        log_clear_names = False
+        idf = load_idf(idf_file)
+
+        # Clean names of idf objects (e.g. 'MATERIAL')
+        idf_2 = deepcopy(idf)
+        clear_name_idf_objects(idf_2, log_clear_names)
+
+        # Get objects from IDF file
+        (
+            buildingSurfs,
+            buildings,
+            constructions,
+            equipments,
+            fenestrationSurfs,
+            globGeomRules,
+            lights,
+            locations,
+            materialAirGap,
+            materialNoMass,
+            materials,
+            peoples,
+            versions,
+            zones,
+            zonelists,
+        ) = get_idf_objects(idf_2)
+
+        surface_init = deepcopy(buildingSurfs[0])
+        _relative_to_absolute(buildingSurfs[0], 1, 2, 3)
+
+        # Asserts relative coords converted to absolute ones
+        assert (
+            buildingSurfs[0]["Vertex_" + str(1) + "_Xcoordinate"]
+            == surface_init["Vertex_" + str(1) + "_Xcoordinate"] + 1
+        )
+        assert (
+            buildingSurfs[0]["Vertex_" + str(1) + "_Ycoordinate"]
+            == surface_init["Vertex_" + str(1) + "_Ycoordinate"] + 2
+        )
+        assert (
+            buildingSurfs[0]["Vertex_" + str(1) + "_Zcoordinate"]
+            == surface_init["Vertex_" + str(1) + "_Zcoordinate"] + 3
+        )
+
+    def test_save_t3d(self, config, converttesteasy):
+        output_folder = None
+        (
+            idf,
+            idf_file,
+            weather_file,
+            window_lib,
+            trnsidf_exe,
+            template,
+            _,
+        ) = converttesteasy
+        try:
+            (
+                idf_file,
+                weather_file,
+                window_lib,
+                output_folder,
+                trnsidf_exe,
+                template,
+            ) = _assert_files(
+                idf_file, weather_file, window_lib, output_folder, trnsidf_exe, template
+            )
+        except:
```

Comment:
This except statement is too general. try, except what? KeyError? ValueError?

Reply:
Even better, put the `_assert_files` under a `with pytest.raises(Exception):` and get rid of the try, except clause

Make sure to replace "Exception" by the actual exception your method raises. This way, pytest will be happy and display the correct information

Reply:
See https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions for more info